h2o-algos/src/main/java/hex/pca/PCA.java

```diff
@@ -314,18 +309,23 @@ public void computeImpl() {
           // Compute SVD of Gram A'A/n using JAMA library
           // Note: Singular values ordered in weakly descending order by algorithm
           _job.update(1, "Calculating SVD of Gram matrix locally");
-          Matrix gramJ = _wideDataset ? new Matrix(ogtsk._gram.getXX()) : new Matrix(gtsk._gram.getXX());
-          SingularValueDecomposition svdJ = gramJ.svd();
+          DenseMatrix gramJ = _wideDataset ? new DenseMatrix(ogtsk._gram.getXX()) : new DenseMatrix(gtsk._gram.getXX());
+          int gramDimension = gramJ.numRows();
+          no.uib.cipr.matrix.SVD svdJ;
+          try {
+            // Note: gramJ will be overwritten after this
+            svdJ = new no.uib.cipr.matrix.SVD(gramDimension, gramDimension).factor(gramJ);
```

Comment:
Hmm, interesting! I just tried the first working solution, without making it too neat yet. Off the top of my head

* a class with wrapper SVD methods, each from a different library 
* a general `SVDSolver` interface, and then classes implemented this interface. Each would use one version / library of SVD decomposition

I'd personally prefer the second solution.

If you want, I can implement that in the future. Just let me know :smile: